### PR TITLE
Update sample plugin to make compatible with vnext

### PR DESCRIPTION
**Pull request by ATorrise**
**What It Does**
- [x] Update imports and resolve errors such that the sample CLI plug-in can be used by plug-in developers to create V3-compatible Zowe CLI plug-ins.
- [x] Restructures and updates the readme.
- [x] Removes any references to jenkins or jenkins files.
- [x] Ensure that you can build and install the plugin as a plugin using zowe cli 8.x & confirm that sample plugin command group is accessible.

**How to Test**
Test that this branch can build a working plugin that runs the sample command `zowe zowe-cli-sample list directory-contents`

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,11 @@ All notable changes to the Sample Plug-in for Zowe CLI will be documented in thi
 
 ## Recent Changes
 
-- Enhancement: Added plugin lifecycle example.
+- Enhancement:
```
Comment (anaxceron) [inline CHANGELOG.md:7]: I would add pr or issue links to both of these enhancement items

**Review by anaxceron — APPROVED** [review]
Left one suggestion -- lookin' good, thank you @ATorrise

File: package.json [hunk 2]
```diff
@@ -83,7 +83,7 @@
     "uuid": "^3.4.0"
   },
   "peerDependencies": {
-    "@zowe/imperative": "^5.13.1"
+    "@zowe/imperative": "^8.0.0-next"
```
Comment (zFernand0) [inline package.json:86]: should we do a range of `^8.0.0-next.0 <8.0.0` (or `^8.0.0-next.0 <9.0.0` to allow `8.x`)

File: docs/tutorials/profile-example/ProfilePlugin.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Implementing profiles in a Zowe CLI plug-in
+# Implementing profiles in a Zowe CLI Plug-in
```
Comment (zFernand0) [inline docs/tutorials/profile-example/ProfilePlugin.md:1]: should plug-in be lowercase?

File: docs/tutorials/list-typicode-todo/ListTypicodeTodoPlugin.md [hunk 4]
```diff
@@ -161,7 +161,7 @@ is presented. You can also see how to use your optional `--id` flag:
 ![output](images/completedSample.png)
 
 # Summary
-You extended an existing Zowe CLI plug-in by introducing a Node.js programmatic API, and you created a command definition with a handler.
+You extended an existing Zowe CLI Plug-in by introducing a Node.js programmatic API, and you created a command definition with a handler.
```
Comment (zFernand0) [inline docs/tutorials/list-typicode-todo/ListTypicodeTodoPlugin.md:164]: same as previous comment. 
should it be lowercase `p`

File: src/pluginDef.ts [hunk 5]
```diff
@@ -13,8 +13,8 @@ import { ProfileConstants } from "./api/ProfileConstants";
 
 const pluginDef: IImperativeConfig = {
     commandModuleGlobs: ["**/cli/*/*.definition!(.d).*s"],
-    pluginLifeCycle: __dirname + "/LifeCycleForSample",
     pluginSummary: "Zowe CLI sample plug-in",
+    pluginLifeCycle: "LifeCycleForSample",
```
Comment (zFernand0) [inline src/pluginDef.ts:17]: I haven't tested this but I'm curious if we need to reference this file from its transpiled location (`lib/LifeCycleForSample`)

File: docs/tutorials/list-directory-contents/ListDirectoryContentsPlugin.md [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
 
-Before you begin, [setup](../Setup.md) your environment to install a plug-in.
+(NOTE: This step may have completed during [setup](../Setup.md).)
+
+Before you begin, ensure you've completed the [initial setup](../../../README.md#initial-setup) of your environment to install a plug-in.
 
 ## Overview
 
-This tutorial covers installing and running this bundled Zowe CLI plugin as-is (without modification), which will display your current directory contents.
+This tutorial covers running this bundled Zowe CLI Plug-in as-is (without modification), which will display your current directory contents.
```
Comment (zFernand0) [inline docs/tutorials/list-directory-contents/ListDirectoryContentsPlugin.md:9]: `plug-in` ?
cc @anaxceron

File: docs/tutorials/list-directory-contents/ListDirectoryContentsPlugin.md [hunk 7]
```diff
@@ -88,13 +88,18 @@ mode  size   birthed                                                   lastModif
...
-Issue the following command:
+#### Setup your system connection details:
+1. `cd __tests__/__resources__/properties`
+2. Copy `example_properties.yaml` to `custom_properties.yaml`.
+3. Edit the properties within `custom_properties.yaml` to contain valid system information for your site.
 
-* `npm run test`
+#### Run tests:
+4. `cd` to your `zowe-cli-sample-plugin` folder
+5. `npm run test`
```
Comment (zFernand0) [inline docs/tutorials/list-directory-contents/ListDirectoryContentsPlugin.md:102]: should the list start from `1.`?

If the intention is for people to setup their connection details, maybe we need a small sentence right before these steps 😋

File: README.md [hunk 8]
```diff
@@ -1,88 +1,87 @@
...
-
-### Install the zowe-cli-sample-plugin to Zowe CLI
+1. `cd` to your `zowe-tutorial` folder
+2. `git clone https://github.com/zowe/zowe-cli-sample-plugin`
+3. `cd zowe-cli-sample-plugin`
+4. `npm install`
+5. `npm run build`
 
-This process assumes that you already installed Zowe CLI on your PC in the previous steps.
+## Next steps: Using your Sample Plug-in
```
Comment (zFernand0) [inline README.md:85]: Do we capitalize every word in titles?
I remember having to fix this in other plug-ins 😋 
cc @anaxceron

File: README.md [hunk 9]
```diff
@@ -1,88 +1,87 @@
...
+  - [Enhancing the sample](docs/tutorials/files-util/FilesUtilPlugin.md) with a third-party package and creating a new command for file-diff highlighting.
+  - [Augmenting the code](docs/tutorials/list-typicode-todo/ListTypicodeTodoPlugin.md) to connect with Typicode's external API, a simulator of backend data, to walk through the cli's interaction with RESTful services.
+  - [Using profiles](docs/tutorials/profile-example/ProfilePlugin.md) with your plug-in.
+- Examples of lifecycle hooks and GitHub Actions workflows for CICD.
+- Templates for unit, system and integration testing suites.
 
 ## Why Create a Zowe CLI Plug-in?
 
-You might want to create a Zowe CLI plug-in to accomplish the following:
+You might want to create a Zowe CLI Plug-in to accomplish the following:
```
Comment (zFernand0) [inline README.md:32]: There may be more instances of `Plug-in` that may need to go back to lowercase, unless we consider the word plug-in as part of the name? (@anaxceron)

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 
left a few questions/comments

File: CHANGELOG.md (hunk 1 above)
Comment (ATorrise) [reply CHANGELOG.md:7]: good catch thank you!!

File: docs/tutorials/profile-example/ProfilePlugin.md (hunk 3 above)
Comment (ATorrise) [reply docs/tutorials/profile-example/ProfilePlugin.md:1]: I think profiles should be capitalized actually

File: src/pluginDef.ts (hunk 5 above)
Comment (ATorrise) [reply src/pluginDef.ts:17]: tests were breaKing without this line @t1m0thyj

File: docs/tutorials/list-directory-contents/ListDirectoryContentsPlugin.md (hunk 7 above)
Comment (ATorrise) [reply docs/tutorials/list-directory-contents/ListDirectoryContentsPlugin.md:102]: @anaxceron what do you think about this?

File: README.md (hunk 8 above)
Comment (ATorrise) [reply README.md:85]: i dont think words like `your` need capitalization but idk

File: src/pluginDef.ts (hunk 5 above)
Comment (t1m0thyj) [reply src/pluginDef.ts:17]: It seems safer to keep the absolute path like we had before:
`pluginLifeCycle: __dirname + "/LifeCycleForSample",`
But if it works with a relative path then maybe that's fine too? 😋

File: README.md (hunk 9 above)
Comment (ATorrise) [reply README.md:32]: made the fix. thanks fernando!

File: package.json [hunk 10]
```diff
@@ -58,7 +58,7 @@
     "@typescript-eslint/parser": "^5.59.5",
     "@zowe/cli": "^7.23.2",
     "@zowe/cli-test-utils": "^7.23.2",
-    "@zowe/imperative": "^5.22.2",
+    "@zowe/imperative": "^8.0.0-next < 9.0.0",
```
Comment (zFernand0) [inline package.json:61]: your initial version string was accurate.
After double-checking on https://semver.npmjs.com/ 
if you do `@zowe/imperative` and `^5.0.0-next` you should only get all 5.x (with prerelease versions included)

**Review by zFernand0 — COMMENTED** [review]
thanks for being so quick to address the comments,
will re-approve once the build is ok 😋

File: README.md (hunk 8 above)
Comment (anaxceron) [reply README.md:85]: @zFernand0 @ATorrise -- I don't think so b/c they're specific commands. You could say "Enter `blah` command to change your directory to `abc`. Or: Issued `blah` to clone the `plugin` repository.

File: docs/tutorials/list-directory-contents/ListDirectoryContentsPlugin.md (hunk 7 above)
Comment (anaxceron) [reply docs/tutorials/list-directory-contents/ListDirectoryContentsPlugin.md:100]: @zFernand0 @ATorrise -- Sorry, missed these two things: 

- In Line 93: **Set up** should be two words b/c it's a verb
- The **Run tests:** list should start w/ Step 1

If you're looking to make one list w/ mini lists, do this:
1. Set up your connection details
   1. cd
   2. copy ...
   3. edit ...
2. Run tests:
   1. cd
   2. npm run test

File: README.md (hunk 8 above)
Comment (ATorrise) [reply README.md:85]: fixing all titles

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -14,20 +14,20 @@ The following code block should be inserted into the Changelog above the last re
...
 | Guidelines for running tests on Zowe CLI | [Testing Guidelines](https://github.com/zowe/zowe-cli/blob/master/docs/TESTING.md) |
 | Guidelines for running tests on the plug-ins that you build for Zowe CLI | [Plug-in Testing Guidelines](https://github.com/zowe/zowe-cli/blob/master/docs/PluginTESTINGGuidelines.md) |
 | Documentation that describes the features of the Imperative CLI Framework | [About Imperative CLI Framework](https://github.com/zowe/imperative/wiki) |
-Versioning conventions for Zowe CLI and Plug-ins| [Versioning Guidelines](https://github.com/zowe/zowe-cli/blob/master/docs/MaintainerVersioning.md) |
+Versioning conventions for Zowe CLI and plug-ins| [Versioning Guidelines](https://github.com/zowe/zowe-cli/blob/master/docs/MaintainerVersioning.md) |
 
-## Contribution Guidelines Specific to the CICS Plug-in
-The following guidelines apply specifically to the Sample Plug-in for Zowe CLI:
+## Contribution guidelines specific to the CICS plug-in
+The following guidelines apply specifically to the Sample plug-infor Zowe CLI:
```
Comment (zFernand0) [inline CONTRIBUTING.md:31]: space 😋

File: README.md [hunk 12]
```diff
@@ -1,88 +1,87 @@
-# Zowe CLI Sample Plug-in
+# Zowe CLI sample plug-in
 
-This repository contains a sample Zowe CLI plug-in that adheres to the contribution guidelines for the project. Use this project and the associated tutorials as a starting point for creating Zowe CLI plug-ins.
+This repository contains a sample Zowe CLI plug-in that adheres to the contribution guidelines for the project. Use this project and the associated tutorials as a starting point for creating Zowe CLI Plug-ins.
```
Comment (zFernand0) [inline README.md:3]: plug-ins (lowercase)

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,10 +1,14 @@
 # Changelog
 
-All notable changes to the Sample Plug-in for Zowe CLI will be documented in this file.
+All notable changes to the Sample plug-in for Zowe CLI will be documented in this file.
```
Comment (zFernand0) [inline CHANGELOG.md:3]: is `Sample` a name? 
if so, I agree with the caps (and we should probably capitalize it in other places)
if not, maybe we should change it to lowercase 😋

File: docs/tutorials/files-util/FilesUtilPlugin.md [hunk 14]
```diff
@@ -2,12 +2,12 @@
 Before you begin this tutorial, make sure that you completed the [Extending an existing plug-in]() tutorial.
 
 ## Overview
-This tutorial demonstrates creating a brand new Zowe CLI plug-in that uses Zowe CLI Node.js programmatic APIs.
+This tutorial demonstrates creating a brand new Zowe CLI plug-inthat uses Zowe CLI Node.js programmatic APIs.
```
Comment (zFernand0) [inline docs/tutorials/files-util/FilesUtilPlugin.md:5]: space?

File: docs/tutorials/list-typicode-todo/ListTypicodeTodoPlugin.md [hunk 15]
```diff
@@ -161,7 +161,7 @@ is presented. You can also see how to use your optional `--id` flag:
 ![output](images/completedSample.png)
 
 # Summary
-You extended an existing Zowe CLI plug-in by introducing a Node.js programmatic API, and you created a command definition with a handler.
+You extended an existing Zowe CLI plug-inby introducing a Node.js programmatic API, and you created a command definition with a handler.
```
Comment (zFernand0) [inline docs/tutorials/list-typicode-todo/ListTypicodeTodoPlugin.md:164]: space?

File: package.json [hunk 16]
```diff
@@ -58,7 +58,7 @@
     "@typescript-eslint/parser": "^5.59.5",
     "@zowe/cli": "^7.23.2",
     "@zowe/cli-test-utils": "^7.23.2",
-    "@zowe/imperative": "^5.22.2",
+    "@zowe/imperative": "^5.0.0-next",
```
Comment (zFernand0) [inline package.json:61]: ^8.0.0-next

File: package.json [hunk 17]
```diff
@@ -83,7 +83,7 @@
     "uuid": "^3.4.0"
   },
   "peerDependencies": {
-    "@zowe/imperative": "^5.13.1"
+    "@zowe/imperative": "^5.0.0-next"
```
Comment (zFernand0) [inline package.json:86]: ^8.0.0-next

File: src/cli/list/profile-args/ProfileArgs.handler.ts [hunk 18]
```diff
@@ -47,9 +47,6 @@ export default class ProfileArgsHandler extends ListBaseHandler {
         if (usingTeamConfig) {
```
Comment (zFernand0) [inline src/cli/list/profile-args/ProfileArgs.handler.ts:47]: we could remove the `usingTeamConfig` logic for v3 😋

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 
left a few comments, but so far so good 😋

File: CHANGELOG.md (hunk 13 above)
Comment (ATorrise) [reply CHANGELOG.md:3]: lowercase! ty

**Review by adam-wolfe — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]
Thanks for addressing the feedback so quickly 🥳 

And also for allowing the vNext prerelease to go out 😋 

LGTM! 😋

**Pull request merged**